### DISCUSSION TITLE
Add a recycle bin to Umbraco Forms

### DISCUSSION THREAD

**Discussion by NikRimington** [umbraco/Umbraco.Forms.Issues#1506] [Ideas]
## Background
We've recently had an incident where a client deleted multiple forms from the Umbraco back office by accident. These were using a version of Forms that stores them in the database. We were then asked to restore these forms several days later which we were "unable" to do easily (database restores and insertion scripts and the like).

## Feature Idea

It would be great if Forms had a recycle bin where forms could be restored from. This would bring managing deletions for forms to a similar point as Content and Media

**Comment by AaronSadlerUK** [comment]
+1

**Comment by emilybotha** [comment]
+1
